src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/RestoreAzureRMRecoveryServicesBackupItem.cs

```diff
@@ -253,6 +251,41 @@ public class RestoreAzureRmRecoveryServicesBackupItem : RSBackupVaultCmdletBase
         [ValidateNotNullOrEmpty]
         public SwitchParameter RestoreAsManagedDisk { get; set; }
 
+        /// <summary>
+        /// Rehydration priority for Archive recovery points.
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = AzureVMParameterSet,
+            HelpMessage = ParamHelpMsgs.RecoveryPoint.RehydratePriority)]
+        [Parameter(Mandatory = false, ParameterSetName = AzureVMRestoreManagedAsUnmanaged,
+            HelpMessage = ParamHelpMsgs.RecoveryPoint.RehydratePriority)]
+        [Parameter(Mandatory = false, ParameterSetName = AzureVMManagedDiskParameterSet,
+            HelpMessage = ParamHelpMsgs.RecoveryPoint.RehydratePriority)]
+        [Parameter(Mandatory = false, ParameterSetName = AzureVMUnManagedDiskParameterSet,
+            HelpMessage = ParamHelpMsgs.RecoveryPoint.RehydratePriority)]
+        [Parameter(Mandatory = false, ParameterSetName = AzureWorkloadParameterSet,
+            HelpMessage = ParamHelpMsgs.RecoveryPoint.RehydratePriority)]
+        [Parameter(Mandatory = false, ParameterSetName = AzureVMRestoreUnmanagedAsManaged,
+            HelpMessage = ParamHelpMsgs.RecoveryPoint.RehydratePriority)]
+        [ValidateSet("Standard","High")]
+        public string RehydratePriority { get; set; }
+
+        /// <summary>
+        /// Rehydration duration for archived RP. values range from 10 to 30, default value is 15.
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = AzureVMParameterSet,
+            HelpMessage = ParamHelpMsgs.RecoveryPoint.RehydrateDuration)]
+        [Parameter(Mandatory = false, ParameterSetName = AzureVMRestoreManagedAsUnmanaged,
+            HelpMessage =  ParamHelpMsgs.RecoveryPoint.RehydrateDuration)]
+        [Parameter(Mandatory = false, ParameterSetName = AzureVMManagedDiskParameterSet,
+            HelpMessage =  ParamHelpMsgs.RecoveryPoint.RehydrateDuration)]
+        [Parameter(Mandatory = false, ParameterSetName = AzureVMUnManagedDiskParameterSet,
+            HelpMessage =  ParamHelpMsgs.RecoveryPoint.RehydrateDuration)]
+        [Parameter(Mandatory = false, ParameterSetName = AzureWorkloadParameterSet,
+            HelpMessage = ParamHelpMsgs.RecoveryPoint.RehydrateDuration)]
+        [Parameter(Mandatory = false, ParameterSetName = AzureVMRestoreUnmanagedAsManaged,
+            HelpMessage = ParamHelpMsgs.RecoveryPoint.RehydrateDuration)]
+        public string RehydrateDuration = "15";
```

Comment:
This was supposed to be an int

Reply:
Yeah the reason for keeping it a string is It has been defined as a string in the service/SDK and hence to avoid any breaking in future, I kept it as a string. 

Also this is working well in preview for few months now, and I'm doing internal validation for appropriate values  - https://github.com/Azure/azure-powershell/blob/Az.RecoveryServices-preview/src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/RestoreAzureRMRecoveryServicesBackupItem.cs#L312